src/OpenConext/Profile/Entity/AuthenticatedUser.php

```diff
@@ -43,13 +45,47 @@
     /**
      * @param AssertionAdapter $assertionAdapter
      * @param EntityId[] $authenticatingAuthorities
+     *
      * @return AuthenticatedUser
+     * @throws RuntimeException
      */
     public static function createFrom(AssertionAdapter $assertionAdapter, array $authenticatingAuthorities)
     {
+        $attributes = [];
+
+        /** @var Attribute $attribute */
+        foreach ($assertionAdapter->getAttributeSet() as $attribute) {
+            $definition = $attribute->getAttributeDefinition();
+
+            // We need to replace the eduPersonTargetedID attribute value as that is a nested NameID attribute.
+            if ($definition->getName() === 'eduPersonTargetedID') {
+
+                /** @var \DOMNodeList[] $eptiValues */
+                $eptiValues = $attribute->getValue();
+                $eptiDomNodeList = $eptiValues[0];
+
+                if (!$eptiDomNodeList instanceof \DOMNodeList || $eptiDomNodeList->length !== 1) {
+                    throw new RuntimeException(
+                        sprintf(
+                            'EPTI attribute must contain exactly one NameID element as value, received: %s',
+                            print_r($eptiValues, true)
```

Comment:
Nice find! This approach is comparable to [Stepup-Gateway's][gateway], although I'm not sure if PII would be present there. I guess it depends on what type of NameIDs are to be expected.

Regarding the `print_r`:
Whether or not an array of DOMNodeLists is returned is not certain, so what *is* returned would be interesting to note. Furthermore, when a `DOMNodeList` is returned, its length property should be exposed (which is something interesting as well).

We could, instead, note the type of `$eptiDomNodeList` and -- if it is a DomNodeList -- its length. That might be a bit complex for a simple Exception, but would prevent PII (if any) to be leaked.

[gateway]: https://github.com/SURFnet/Stepup-Gateway/blob/383ca16200378de38bac098717ba9ed7a3999e5e/src/Surfnet/StepupGateway/GatewayBundle/Service/ProxyResponseService.php#L203-L207